pkg/workspaces/workspaces.go

```diff
@@ -0,0 +1,224 @@
+		return ws, nil
+	}
+
+	for i := range ws {
+		if v, ok := workspaces[ws[i].Name]; ok {
+			ws[i] = v
+			delete(workspaces, v.Name)
+		}
+	}
+
+	for _, v := range workspaces {
+		ws = append(ws, v)
+	}
+
+	return ws, nil
+}
+
+func parseWorkspace(w []string) (map[string]v1alpha1.WorkspaceBinding, error) {
+	ws := map[string]v1alpha1.WorkspaceBinding{}
+	for _, v := range w {
+
+		r := strings.Split(v, ",")
+		name, err := getPar(r, nameParam)
+		if err != nil {
+			return nil, errors.New("Name not found for workspace")
+		}
+
+		wB := v1alpha1.WorkspaceBinding{
+			Name: name,
+		}
+
+		subPath, err := getPar(r, subPathParam)
+		if err == nil {
+			wB.SubPath = subPath
+		} else if err != nil && err != errNotFoundParam {
+			return nil, err
+		}
+
+		err = setWorkspaceConfig(r, &wB)
+		if err == nil {
+			ws[name] = wB
+			continue
+		} else if err != errNotFoundParam {
+			return nil, err
+		}
+
+		err = setWorkspaceSecret(r, &wB)
+		if err == nil {
+			ws[name] = wB
+			continue
+		}
+
+		err = setWorkspaceEmptyDir(r, &wB)
+		if err == nil {
+			ws[name] = wB
+			continue
+		}
+
+		err = setWorkspacePVC(r, &wB)
+		if err == nil {
+			ws[name] = wB
+			continue
+		}
+		return nil, errors.New(invalidWorkspace + v)
+	}
+	return ws, nil
+}
+
+func setWorkspaceSecret(r []string, wB *v1alpha1.WorkspaceBinding) error {
+	secret, err := getPar(r, secretParam)
+	if err == nil {
+		wB.Secret = &corev1.SecretVolumeSource{
+			SecretName: secret,
+		}
+	}
+	return err
+}
+
+func setWorkspaceConfig(r []string, wB *v1alpha1.WorkspaceBinding) error {
+	config, err := getPar(r, configParam)
+	if err == nil {
+		items, err := getItems(r)
+		if err != nil && err != errNotFoundParam {
+			return err
+		}
+
+		wB.ConfigMap = &corev1.ConfigMapVolumeSource{
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: config},
+			Items: items,
+		}
+	}
+	return err
+}
+
+func getItems(r []string) ([]corev1.KeyToPath, error) {
+	var kp []corev1.KeyToPath
+	for i := range r {
+		if !strings.Contains(r[i], configItemParam) {
+			continue
+		}
+		s := strings.SplitN(r[i], "=", 2)
+		if s[0] != configItemParam {
+			continue
+		}
+		if len(s) != 2 {
+			return nil, errors.New("invalid item")
+		}
+		key, path, err := getKeyValue(s[1])
+		if err != nil {
+			return nil, err
+		}
+		kp = append(kp, corev1.KeyToPath{
+			Key:  key,
+			Path: path,
+		})
+	}
+	return kp, nil
+}
+
+func setWorkspacePVC(r []string, wB *v1alpha1.WorkspaceBinding) error {
+	claimName, err := getPar(r, claimNameParam)
+	if err == nil {
+		wB.PersistentVolumeClaim = &corev1.PersistentVolumeClaimVolumeSource{
+			ClaimName: claimName,
+		}
+	}
+	return err
+}
+
+func getKeyValue(s string) (string, string, error) {
+	r := strings.SplitN(s, "=", 2)
+	if len(r) != 2 {
+		return "", "", errors.New("invalid key value")
+	}
+	return r[0], r[1], nil
+}
+
+func setWorkspaceEmptyDir(r []string, wB *v1alpha1.WorkspaceBinding) error {
+	emptyDir, err := getPar(r, emptyDirParam)
+	if err == nil {
```

Comment:
same as above keep green path closer to left border by returning early 

```
if err != nil {
  return
}

green
path
...
```

Reply:
Just after setting, we want to continue to next iteration.  We aren't returning errors unless it's not found.